<?xml version="1.0" encoding="utf-8"?>
<launch>
    <!-- declare arg to be passed in -->
    <arg name="cad_file" default="$(find cpt_selective_icp)/resources/cla.json" />
    <arg name="lidar_scan" default="$(find cpt_matching_algorithms)/../../data_set/scan200.bag" />
    <arg name="lidar_settings" default="$(find cpt_matching_algorithms)/config/sim_lidar_config.yaml" />

   <node name="test_matcher"  pkg="cpt_matching_algorithms" type="test_matcher_node" output="screen" launch-prefix="tmux split-window -h">
        <rosparam command="load" file="$(find cpt_matching_algorithms)/config/test_match_config.yaml"/>

        <rosparam command="load" file="$(find cpt_matching_algorithms)/config/plane_extraction_config.yaml" />
        
        <param name="goicp_folder" value="$(find go_icp_catkin)"/>
        <rosparam command="load" file="$(find cpt_matching_algorithms)/config/go_icp_config.yaml" />
        
        <rosparam command="load" file="$(arg lidar_settings)" />
        <param name="tfMapFrame" value="/map" />
        <param name="cadTopic" value="/mesh_publisher/mesh_out" />
        <param name="lidarTopic" value="rslidar_points" />
        <param name="simlidarTopic" value="sim_rslidar_points" />
        <param name="groundTruthTopic" value="ground_truth" />
    </node>

    <node name="lidar_simulator"  pkg="cpt_matching_algorithms" type="sim_lidar_node" output="screen" launch-prefix="tmux split-window">
        <rosparam command="load" file="$(arg lidar_settings)" />
        <param name="cadTopic" value="/mesh_publisher/mesh_out" />
    </node>

    <node name="LiDAR_driver" pkg="rosbag" type="play" args="-l $(arg lidar_scan)" />

    <node name="mesh_positioning"
          pkg="cpt_selective_icp"
          type="position_cad_with_marker.py">
        <param name="marker_parent_frame" value="map" />
        <param name="marker_frame_name" value="marker" />
    </node>

    <node name="mesh_publisher" pkg="cpt_utils" type="mesh_publisher_node">
        <param name="publish_on_start" value="false"/>  <!-- if set to true, mesh is published two times -->
        <param name="default_filename" value="$(arg cad_file)"/>
        <param name="frame_name" value="marker"/>
    </node>

    <node pkg="tf" type="static_transform_publisher" name="laser" args="100 0 0 0 0 0 1  /lidar /rslidar  100"/>

    <node name="rviz" pkg="rviz" type="rviz" args="-d $(find cpt_matching_algorithms)/config/rviz.yaml" />

</launch>
